pages/desktop/consumer_pages/base.py

```diff
@@ -68,9 +71,15 @@ def set_region(self, region):
         self.wait_notification_box_not_visible()
 
     def login(self, user=None):
-        from pages.fxa import FirefoxAccounts
-        fxa = FirefoxAccounts(self.testsetup)
-        fxa.login_user(user)
+        credentials = isinstance(user, MockUser) and user or \
+                    self.testsetup.credentials.get(user, FxATestAccount(use_prod=False).create_account())
+        from fxapom.pages.desktop.sign_in import SignIn
+        fxa_login = SignIn(self.testsetup)
+        if user=='default':
+            fxa_login.sign_in(credentials['email'], credentials['password'])
+        elif user=='new':
+            fxa_login.sign_in(credentials.email, credentials.password)
+
```

Comment:
I'm sorry, but this approach really rubs me the wrong way. I don't think that a `login` method should be responsible for creating a test user account. I think the account should be created before calling the login method and then the user can be passed in. Otherwise it's not going to be clear from the test that we are creating a new account - one would have to know the internals of the `login` method to know this.

Also, if we do that we can likely simplify the code in here, which is a bit hard to follow.

@AndreiH and @m8ttyB what do you think about this?